### ISSUE TITLE
Local Ace won't highlight right away

### ISSUE THREAD

**Issue by davidgatti** [ajaxorg/ace#3526]
I have the following situation. I added a new rule to highlight some simple keywords. When I was writing them down one by one, Ace was coloring the words correctly, as seen in the screenshot bellow...

![screen shot 2018-01-23 at 10 38 01 am](https://user-images.githubusercontent.com/411114/35268581-cabefee6-0029-11e8-8848-a114da3219af.png)

And once I had everything, all was nice and blue. But once I saved my work, and restarted Ace, no highlights to be seen. 

![screen shot 2018-01-23 at 10 37 57 am](https://user-images.githubusercontent.com/411114/35268662-025f6b92-002a-11e8-8309-94f1243b87e0.png)

If I delete the regex and start adding regexp rule one by one again, Ace will start coloring the words again. Is this normal behaviour?

**Comment by nightwing** [comment]
Are there any errors in browser console? Does highlighting start to work if you add a space at the end of the file (or do ctrl-a ctrl-c ctrl-v) instead of deleting the regexp.

By the way, is this question related to https://github.com/ajaxorg/ace/pull/3511?

**Comment by davidgatti** [comment]
While writing you a reply I did find the problem, somehow I ended up having the same regexp rule twice. I think the live editor for development is a blessing and a cure at the same time. I think I ended up not hitting save again. 

I think it be nice in the future to add the following to the editor

- Have a banner showing the the changes can't be saved locally, and what to do to enable that.
- Pop a message if the changes are not saved and you try to close the window. 

I think this might minimize any confusion.
